### PR TITLE
Added login route to users

### PR DESCRIPTION
**Pull request by xunchuan-liu**

### CODE REVIEW

File: src/auth.rs [hunk 1]
```diff
@@ -35,7 +36,7 @@ pub async fn validator(
 ) -> Result<ServiceRequest, Error> {
     match credentials.token() {
         "_" => {
-            if req.path() == "/health" {
```
Comment (xunchuan-liu) [inline src/auth.rs:38]: Allow `/login` route without bearer auth

File: src/main.rs [hunk 2]
```diff
@@ -216,6 +216,57 @@ mod tests {
             test::read_response_json(&mut app, req).await;
     }
```
Comment (xunchuan-liu) [inline src/main.rs:218]: Test creating a new user and logging in with that user

File: src/users/model.rs [hunk 3]
```diff
@@ -42,6 +42,12 @@ lazy_static! {
         }
         base64::encode(secret)
     };
```
Comment (xunchuan-liu) [inline src/users/model.rs:44]: Auth seed for bcrypt

File: src/users/model.rs [hunk 4]
```diff
@@ -280,3 +286,40 @@ impl std::convert::TryInto<AuthUser> for User {
         })
     }
 }
```
Comment (xunchuan-liu) [inline src/users/model.rs:288]: Added try_into implementation for MaybeUser struct

File: src/users/routes.rs [hunk 5]
```diff
@@ -55,7 +55,23 @@ async fn create(user: web::Json<MaybeUser>) -> Result<HttpResponse, CustomError>
     Ok(HttpResponse::Ok().json(auth_user))
 }
```
Comment (xunchuan-liu) [inline src/users/routes.rs:57]: New login route

File: src/auth.rs [hunk 6]
```diff
@@ -35,7 +36,7 @@ pub async fn validator(
...
+            if req.path() == "/health" || req.path() == "/login" && req.method() == Method::POST {
```
Comment (trueb2) [inline src/auth.rs:39]:
```suggestion
            if req.path() == "/health" || 
             (req.path() == "/login" && req.method() == Method::POST) {
```

File: src/users/model.rs [hunk 7]
```diff
@@ -241,15 +247,15 @@ impl User {
...
+    fn internal_token(username: String, password: String) -> Result<String, CustomError> {        
         let db_token = bcrypt(password.as_bytes())?;
         let db_token = base64::encode(db_token.as_slice());
         let db_token = format!("$2$10${}", db_token);
         log::trace!(
             "Prepared username: [{}] and password: [{}]",
             username,
             db_token
-        );
+        );
```
Comment (trueb2) [inline src/users/model.rs:258]:
```suggestion
        );
```

File: src/users/model.rs [hunk 8]
```diff
@@ -241,15 +247,15 @@ impl User {
             .first(&conn)?)
     }
 
-    fn internal_token(username: String, password: String) -> Result<String, CustomError> {
+    fn internal_token(username: String, password: String) -> Result<String, CustomError> {
```
Comment (trueb2) [inline src/users/model.rs:250]:
```suggestion
    fn internal_token(username: String, password: String) -> Result<String, CustomError> {
```

**Review by trueb2 — CHANGES_REQUESTED** [review]

**Review by trueb2 — APPROVED** [review]

**Pull request merged**